src/jobsautoreport/slack.py

```diff
@@ -0,0 +1,131 @@
+            {
+                "type": "section",
+                "fields": [
+                    {
+                        "type": "mrkdwn",
+                        "text": f"*Number of jobs triggered*: {self.data['number_of_jobs_triggered']}",
+                    }
+                ],
+            },
+            {
+                "type": "section",
+                "fields": [
+                    {
+                        "type": "mrkdwn",
+                        "text": f"*Average duration*:{self.data['average_jobs_duration']}",
+                    }
+                ],
+            },
+        ]
+
+        return blocks
+
+    def format_as_comment(self) -> list:
+        blocks = [
+            {
+                "type": "header",
+                "text": {
+                    "type": "plain_text",
+                    "text": "Top 10 failing e2e periodic jobs",
+                    "emoji": True,
+                },
+            },
+            {
+                "type": "section",
+                "fields": [
+                    {"type": "mrkdwn", "text": f"1. Job name: {self.data[0].name}"},
+                    {"type": "mrkdwn", "text": f"Job score: {self.data[0].score}"},
+                ],
+            },
+            {
+                "type": "section",
+                "fields": [
+                    {"type": "mrkdwn", "text": f"2. Job name: {self.data[1].name}"},
+                    {"type": "mrkdwn", "text": f"Job score: {self.data[1].score}"},
+                ],
+            },
+            {
+                "type": "section",
+                "fields": [
+                    {"type": "mrkdwn", "text": f"3. Job name: {self.data[2].name}"},
+                    {"type": "mrkdwn", "text": f"Job score: {self.data[2].score}"},
+                ],
+            },
+            {
+                "type": "section",
+                "fields": [
+                    {"type": "mrkdwn", "text": f"4. Job name: {self.data[3].name}"},
+                    {"type": "mrkdwn", "text": f"Job score: {self.data[3].score}"},
+                ],
+            },
+            {
+                "type": "section",
+                "fields": [
+                    {"type": "mrkdwn", "text": f"5. Job name: {self.data[4].name}"},
+                    {"type": "mrkdwn", "text": f"Job score: {self.data[4].score}"},
+                ],
+            },
+        ]
+
+        return blocks
+
+
+class SlackMessenger:
+    def __init__(self, web_client: WebClient) -> None:
+        self.client = web_client
+
+    def send_message(self, msg: SlackMessage) -> None:
+        blocks_list = msg.format_as_message()
+        try:
+            res = self.client.chat_postMessage(
+                channel=msg.channel_id, blocks=blocks_list
+            )
+            msg.ts = res["ts"]
+            logger.info("message was sent to slack channel %s", msg.channel_id)
+        except SlackApiError as err:
+            assert err.response["error"]
```

Comment:
print a failure message with the exception and raise it back instead. The assert will shadow the type of the exception which can be an issue if we want to act it (e.g.: we want to send the message again).

https://docs.python.org/3/tutorial/errors.html#handling-exceptions